src/models/github-server-app.test.ts

```diff
@@ -128,6 +135,92 @@ describe("GitHubServerApp", () => {
+				const originalPrivateKey = "privatekey";
+				let gitHubServerApp;
+
+				beforeEach(async () => {
+					gitHubServerApp = await GitHubServerApp.create({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: clientIdWithTypo,
+						gitHubClientSecret: "secret",
+						webhookSecret: "anothersecret",
+						privateKey: "privatekey",
+						installationId
+					});
+				});
+
+				it("should update GitHub app when uuid is found", async () => {
+					const originalApp = await GitHubServerApp.findForUuid(uuid);
+					expect(originalApp && originalApp.gitHubClientId).toEqual(clientIdWithTypo);
+					expect(originalApp && originalApp.webhookSecret).toEqual(`encrypted:${originalWebhookSecret}`);
+					expect(originalApp && originalApp.privateKey).toEqual(`encrypted:${originalPrivateKey}`);
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(uuid);
+					expect(updatedApp && updatedApp.uuid).toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && await updatedApp.decrypt("webhookSecret")).toEqual("anewsecret");
+					expect(updatedApp && await updatedApp.decrypt("privateKey")).toEqual("privatekeyversion2");
+				});
+
+				it("should not update GitHub app when uuid is not found", async () => {
+					const mismatchedUUID = "c97806fc-c433-4ad5-b569-bf5191590ba9";
+
+					await GitHubServerApp.updateGitHubAppByUUID({
+						uuid: mismatchedUUID,
+						appId: 1,
+						gitHubAppName: "my awesome app",
+						gitHubBaseUrl,
+						gitHubClientId: "lvl.1n23j12389wnde",
+						gitHubClientSecret: "secret",
+						webhookSecret: "anewsecret",
+						privateKey: "privatekeyversion2",
+						installationId
+					});
+
+					const updatedApp = await GitHubServerApp.findForUuid(mismatchedUUID);
+					expect(updatedApp && updatedApp.uuid).not.toEqual(gitHubServerApp.uuid);
+					expect(updatedApp && updatedApp.gitHubClientId).not.toEqual("lvl.1n23j12389wnde");
+					expect(updatedApp && updatedApp.webhookSecret).not.toEqual("anewsecret");
+					expect(updatedApp && updatedApp.privateKey).not.toEqual("privatekeyversion2");
+				});
+
+				it("should only update values changed and leave other values as is", async () => {
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					await GitHubServerApp.updateGitHubAppByUUID({ uuid, appId: 1, gitHubBaseUrl, webhookSecret: "updatedSecret" });
+					const myApp = await GitHubServerApp.findForUuid(uuid);
+
+					expect(myApp && myApp.uuid).toEqual(uuid);
```

Comment:
Btw, I think we can change all these `expect(xxx & xxx.gitHubAppName)...` to just `expect(xxx.gitHubAppName)`. 
The `myApp` here has to be not null, if null, test should fail, so it is okay to skip the `myApp &&`

Reply:
The test won't fail because in these tests the result **isn't** null. However, because the GHSA methods either return an app (if found) or null e.g.
```
	static async findForUuid(uuid: string): Promise<GitHubServerApp | null> {
		return this.findOne({
			where: {
				uuid
			}
		});
	}
```

we get a TS error if we don't confirm there is in fact an a

![Screen Shot 2022-08-12 at 12 16 25 pm](https://user-images.githubusercontent.com/37155488/184272708-2190ef5a-0f74-4994-a280-337ee5351a1d.png)

So we can't simply remove this. I can tidy this up a little with optional chaining though.